views/js/runner/plugins/controls/progressbar/progressbar.js

```diff
@@ -136,6 +137,103 @@ define([
                 var total = counter.total();
                 var currentPosition = counter.position();
 
+                return {
+                    ratio : total > 0 ? Math.floor(currentPosition / total * 100) : 0,
+                    label : progressConfig.showTotal ?
+                        __('Item %d of %d', currentPosition, total) :
+                        __('Item %d', currentPosition)
+                };
+            },
+
+            /**
+             * Updates the progress bar displaying the questions only
+             * ( for example informational items won't be counted and will be without progress bar)
+             * @param {Object} testContext The progression context
+             * @returns {{ratio: number, label: string}}
+             */
+            questions : function position() {
+
+                //get the current test part in the map
+                var getTestPart = function getTestPart(){
+                    if(testMap && testMap.parts){
+                        return testMap.parts[testContext.testPartId];
+                    }
+                };
+
+                //get the current test section in the map
+                var getTestSection = function getTestSection() {
+                    var testPart = getTestPart();
+                    if(testPart && testPart.sections){
+                        return testPart.sections[testContext.sectionId];
+                    }
+                };
+
+                //provides you the methods to get total and position by scope
+                var progressScopeCounter = {
+                    test : {
+                        total : function(){
+                            if (testMap && testMap.parts) {
+                                return _.reduce(testMap.parts, function (accPart, part) {
+                                    return _.reduce(part.sections, function (accSection, section) {
+                                        return accSection + mapHelper.computeItemStats(section.items).questions;
+                                    }, accPart);
+                                }, 0);
+                            }
+                            return 0;
+                        },
+                        position : function(){
+                            if (testMap && testMap.parts) {
+                                return _.reduce(testMap.parts, function (accPart, part) {
+                                    return _.reduce(part.sections, function (accSection, section) {
+                                        return accSection + mapHelper.computeItemStats(section.items).questionsViewed;
```

Comment:
I foresee potential performance issue here, instead of using this kind of construction, couldn't you use the stats directly?